### ISSUE TITLE
Select a Person Responsible for the report as part of Operation Information Review

### ISSUE THREAD

**Issue by suhafa** [bcgov/cas-reporting#255]
#### Description:
"As an Industry Reporter, I would to confirm or add contact information for the Person Responsible so that this information is accurate and included in the report"

We need a screen under/after the 'Review Operation Information' section, that allows user to assign 'Person Responsible for preparing and submitting the report' so that there is someone identified as the main contact for the report. This can be a list of pre-existing users already specified under the 'Contacts' page elsewhere in our app, and so we need a searchable drop down populated with a list of all the contacts in BCIERS.

[Wireframe here](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=21271-4976&t=w5hKibTjTD9PMbWt-4)

#### Acceptance Criteria:

Given that I am an Industry Reporter
When I am reviewing my operation information
Then I need a step where I select the person responsible for preparing and submitting a report

Given that I am in Industry Reporter
When I'm on the the form to select the person responsible for the report
Then I want to select a person from the contact list filled out in registration ~, or add a new contact to select as person responsible for the report.~

Given I am an industry reporter
When select a person responsible for my report
Then I can see its info added (read-only) to the page

Given I am an industry reporter
When select a person responsible for my report
Then I can only select one person

#### Development Checklist:

- [x] Add a page under 'Review Operation Information' with a searchable dropdown for contacts.
- [x] Fetch contacts from **erc.contact** and populate the dropdown.
- [x] Allow users to manually enter contact information if not selecting from the dropdown.
- [x] Implement fields for First Name, Last Name, Job Title/Position, Business Email Address, Business Phone Number, and Business Mailing Address.
- [x] Add form validation to ensure all required fields are filled correctly.
- [x] "Continue" button from operation review page goes to the Person Responsible page
- [x] Card for "Sync with database" button(https://github.com/bcgov/cas-reporting/issues/364)
- [x] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by patriciarussellCAS** [comment]
Ticket just needs Dev Checklist. Can a dev tackle this? @joshgamache @ayeshmcg @BallardRobinett

**Comment by rdromey** [comment]
@ayeshmcg to write dev checklist

**Comment by rdromey** [comment]
Need to confirm contact management process with the BA for reports.

**Comment by pbastia** [comment]
BA: We will be copying the contact information into the report (per regulations)

**Comment by patriciarussellCAS** [comment]
Additional follow-up considerations: 
After a person has been selected in the Report, this will be saved in the Information Management module (meaning that you'll be able to see that a contact has been assigned as "person responsible for submitting a report" on the contact page in Info Management). We currently don't have a workflow or a place where this information can be edited/changed later. Do we want them to come back into the reporting workflow to update this? TBD

**Comment by pbastia** [comment]
@patriciarussellCAS @suhafa @dleard marking this as blocked until a few things are resolved with the BA, and circling back with @andrea-williams to make sure we're consistent with what's been done in registration contact selection

**Comment by andrea-williams** [comment]
Taking a quick look at this ticket and the wireframes, it seems pretty consistent with what we're doing in Reg/Admin, except for the mailing address field for a new contact. We've separated the address section out into separate fields for street address, municipality, province, and postal code. Those are all stored as separate cols in the DB so it's easier to collect the info that way rather than writing code to parse all that info out from just 1 string.

**Comment by patriciarussellCAS** [comment]
My thoughts on this ticket are to do things the same way Moose are doing it for Operation Representative in Registration. 
So users can selected a "person responsible" from the dropdown list of Contacts. If they need to add a contact they can do this on the page and then the information is saved in contacts. We can separate the mailing address the same way too if that makes sense. 
@pbastia do you have any outstanding questions or do we have enough information to unblock this?

**Comment by pbastia** [comment]
That makes sense! Thanks @patriciarussellCAS

**Comment by pbastia** [comment]
@patriciarussellCAS AC have been updated with the results of discussions with BA. Would you mind reviewing it for a sanity check?

**Comment by patriciarussellCAS** [comment]
@zoeyli-46 Ayesha has built this page so that it has the same layout as Operation Representative in registration. I don't think this is reflected in the wireframes yet. Not a big priority but something to clean up at some point :)
Thanks!

**Comment by zoeyli-46** [comment]
@patriciarussellCAS done! updated the wireframe [link](https://www.figma.com/design/LsDmLDCdnJqI8UahyqvOD8/BCIERS-Designs?node-id=21271-4976&t=w5hKibTjTD9PMbWt-4) in this ticket.

**Comment by ayeshmcg** [comment]
@zoeyli-46  This is the sample
<img width="739" alt="image" src="https://github.com/user-attachments/assets/ef651d36-abe8-405b-a230-83570d7e2191">
As per the discussion the contact will be added/edited from administration/contact page, this will display read-only contact details(but in the format as it is displayed in registration)

**Comment by ayeshmcg** [comment]
To test:-
Go to https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/reporting/reports
click on start for any operation-> click on Save and continue(on 
Review operation information page) -> This will take you to Person Responsible Page
or https://cas-bciers-frontend-dev.apps.silver.devops.gov.bc.ca/reporting/reports/4/person-responsible
